### ISSUE TITLE
Jsonnet examples and documentation don't seem to work on k `1.22`

### ISSUE THREAD

**Issue by zalegrala** [grafana/mimir#1825]
#### Describe the bug

I'm following the docs to create a new mimir deployment in a lab environment and running into an issue.

#### To Reproduce

1. `cp vendor/mimir/mimir-manifests.jsonnet.example environments/mimir/main.jsonnet`
2. `tk diff environments/mimir`
Observe the following error.

```
Error: evaluating jsonnet: RUNTIME ERROR: max stack frames exceeded.
	Field "#mixin"	
	Field "mixin"	
	Field "mixin"	
	Field "mixin"	
	Field "mixin"	
	Field "mixin"	
	Field "mixin"	
	Field "mixin"	
	Field "mixin"	
	Field "mixin"	
	...
	Field "nodeAffinity"	
	Field "affinity"	
	Field "spec"	
	Field "template"	
	Field "spec"	
	Field "daemonSet"	
	Field "v1beta1"	
	Field "extensions"	
	Field "consul"	
	During manifestation	
```

#### Expected behavior

I'd expect that the objects render and give back the delta between the running namespace, and the rendered manifest.

#### Environment

- Infrastructure: kubernetes on k3s
- Deployment tool: tanka + jsonnet

#### Additional Context

Of note, I'm using `github.com/jsonnet-libs/k8s-libsonnet/1.22/main.libsonnet` instead of the `1.18` as mentioned in the quickstart.

**Comment by zalegrala** [comment]
I'm happy to submit a PR if someone could point me in the right direction.

**Comment by pracucci** [comment]
Does the `operations/mimir/getting-started.sh` work to you? If yes, then I would investigate backward from there. If not, do you get the same error reported in the issue description or a different one?

**Comment by zalegrala** [comment]
Ah, thank you for the pointer.  It does appear to be related to `1.22` that I'm using for `k`.  The `1.18` looks to work.  I'll update the issue title.

**Comment by zalegrala** [comment]
`1.21` also looks to work fine.

**Comment by pracucci** [comment]
I took a quick look and I haven't found the root cause yet. I think we're having an infinite recursions and I've noticed it caused by `legacy-noname.libsonnet` vendored via `ksonnet-util`.

**Comment by andyasp** [comment]
Saw this in the backlog and went to see if anything has changed. I think this issue has been fixed by https://github.com/grafana/jsonnet-libs/pull/800.

**Comment by zalegrala** [comment]
Thanks for the ping, this appears to be resolved in 1.22 now.
